### DISCUSSION TITLE
[feature request] Windows builds on Chocolatey

### DISCUSSION THREAD

**Discussion by moghingold** [input-leap/input-leap#1737] [Feature Requests]
### What happened?

Barrier no longer connects on Windows, so I need to switch to input-leap since it is actively maintained. However, the released tags for input-leap are not newer than the ones for Barrier, so there is no signed input-leap binary I can install on Windows.

Expected result:
Input-leap should schedule Windows builds with incremental version numbers and make the installer available to Chocolatey.

Current result:
I can technically follow the (outdated) build instructions in the Wiki to create my own debug build of input-leap if I have a lot of hard drive space, and a lot of time.

### Version

v2.4.0

### What OSes are you seeing the problem on? (Check all that apply)

Windows

### What OS versions are you using?

Windows 10 (Barrier 2.4.0 client)
Arch GNU/Linux (Barrier 2.4.0 server)

**Comment by shymega** [comment]
We don't know yet if we'll be doing signed builds for Windows (this goes for Mac too), but v3.0.0 will be released soon, and signed builds would be good to have. Keeping this issue open for Chocately, we have another for Scoop.

**Comment by TheAkashicTraveller** [comment]
Just FYI, Barrier can work on windows. You need to either disable SSL, in both barrier clients or use openSSL to manually generate the SSL certificate it needs, easiest way to get it on windows is by installing git for windows. There's an issue in the Barrier github about it that kind of explains it but no one summarised it.

To generate the certificate using CMD enter `cd %userprofile%\AppData\Local\Barrier\SSL` and `"C:\Program Files\Git\usr\bin\openssl.exe" req -x509 -nodes -days 3365 -subj /CN=Barrier -newkey rsa:4096 -keyout Barrier.pem -out Barrier.pem -sha256`

Note this is a fix for an issue with barrier not input leap and doesn't necessarily apply here. Though if you builf 2.4.0 for windows and the same issue shows up, that being `ERROR: ssl certificate doesn't exist:` in the log, then it may help. Not sure why anyone would do that now that 3.0.0 seems to be on the horizon.

**Comment by shymega** [comment]
This isn't Barrier though, and so fixes *for* Barrier don't necessarily apply here. I would edit your comment to reflect that difference.

**Comment by ylluminate** [comment]
I couldn't care less about signed versions here personally so I might suggest we start simply without signed versions for simplicity and then perhaps revisit it later if demand necessitates.

**Comment by shymega** [comment]
> I couldn't care less about signed versions here personally so I might suggest we start simply without signed versions for simplicity and then perhaps revisit it later if demand necessitates.

It might not be an issue for you, personally, but for work computers, we've had requests for signed versions for macOS and Windows. This is because of things like SCCM and Jamf requiring signed executables. Once the CI PR is merged, we can explore building for Apple Silicon, Qt 6, and signed builds.

**Comment by ylluminate** [comment]
@shymega I understand what you're saying and I apologize if this came off brash... I was attempting to lower the entry bar for getting the ball rolling vs having to resolve it initially. It seems as though it is a matter of simplifying the situation.

Also, fwiw, I admin a lot of work workstations on small to mid-sized businesses and usually signing is not that big of a deal, but you're right, when you get into upper-mid to enterprise it does become an issue.

**Comment by shymega** [comment]
Sure, no worries.

I can't get a signature myself yet, as I don't have the funds to guarantee I can afford it yearly - or at all.

Not sure about @p12tic. We'd need one that works on macOS and Windows. For Linux/BSD, we can use GPG. Probably best to store the secrets in HashiCorp Vault or GH Actions. Not sure yet.

But it is a necessity for future releases, and *ideally* I'd like it to be part of Input Leap for v3.0.0.

**Comment by pklapperich** [comment]
> To generate the certificate using CMD enter `cd %userprofile%\AppData\Local\Barrier\SSL` and `"C:\Program Files\Git\usr\bin\openssl.exe" req -x509 -nodes -days 3365 -subj /CN=Barrier -newkey rsa:4096 -keyout Barrier.pem -out Barrier.pem -sha256`

Thanks! This allowed me to connect Barrier v2.4.0 on windows to InputLeap 2.4.0 on Archlinux

> We'd need one that works on macOS and Windows.

Unfortunately, I'm pretty sure that's not a thing. Someone correct me if this is wrong, but I believe for macOS you need to make an organization in the Apple Developer program and certificates are generated through there. I think that's $99/yr for each developer. I believe Apple is the CA for these certs. For Windows you can get a code signing cert from any of several CAs. The standard certs are about $200/yr and the EV certs are about $300/yr (if you buy multiple years at a time). But getting an EV cert for an organization name (such as "input leap") I think would require making an LLC, corporation, or non-profit. Standard cert is probably good enough, and the warnings at install eventually go away after enough user installs ("reputation"). So I think input-leap is in need of $400-600/yr just for signing certificates.

Maybe this discussion is better continued on IRC, but if you set up a patreon or similar and added a [sponsor button](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/displaying-a-sponsor-button-in-your-repository#:~:text=On%20the%20%22General%22%20settings%20page,the%20instructions%20in%20the%20FUNDING.) in the repo, that might help secure funding for the necessary certificates. I'd hate to see you and @p12tic spending cash on this project; you both already contribute so much of your time!

**Comment by shymega** [comment]
I would rather the discussion continue on the Discussions tab.

Ultimately, I can't see enough funding being secured for both OSes. I see most requests are within the context of an enterprise environment, so it's not out of the question for the employer to sponsor the project.
